src/omnicore/rpc.cpp

```diff
@@ -145,6 +148,318 @@ bool BalanceToJSON(const std::string& address, uint32_t property, Object& balanc
+            "\nResult:\n"
+            "[                       (array of JSON objects)\n"
+            "  {\n"
+            "    \"propertyid\" : nnnnnnn,          (number) the property id\n"
+            "    \"feetrigger\" : \"n.nnnnnnnn\",   (string) the amount of fees required to trigger distribution\n"
+            "  },\n"
+            "  ...\n"
+            "]\n"
+            "\nExamples:\n"
+            + HelpExampleCli("omni_getfeetrigger", "3")
+            + HelpExampleRpc("omni_getfeetrigger", "3")
+        );
+
+    uint32_t propertyId = 0;
+    if (0 < params.size()) {
+        propertyId = ParsePropertyId(params[0]);
+    }
+
+    if (propertyId > 0) {
+        RequireExistingProperty(propertyId);
+    }
+
+    Array response;
+
+    for (uint8_t ecosystem = 1; ecosystem <= 2; ecosystem++) {
+        uint32_t startPropertyId = (ecosystem == 1) ? 1 : TEST_ECO_PROPERTY_1;
+        for (uint32_t itPropertyId = startPropertyId; itPropertyId < _my_sps->peekNextSPID(ecosystem); itPropertyId++) {
+            if (propertyId == 0 || propertyId == itPropertyId) {
+                int64_t feeTrigger = p_feecache->GetDistributionThreshold(itPropertyId);
+                std::string strFeeTrigger = FormatMP(itPropertyId, feeTrigger);
+                Object cacheObj;
+                cacheObj.push_back(Pair("propertyid", (uint64_t)itPropertyId));
+                cacheObj.push_back(Pair("feetrigger", strFeeTrigger));
+                response.push_back(cacheObj);
+            }
+        }
+    }
+
+    return response;
+}
+
+// Provides the fee share the wallet (or specific address) will receive from fee distributions
+Value omni_getfeeshare(const Array& params, bool fHelp)
+{
+    if (fHelp || params.size() > 2)
+        throw runtime_error(
+            "omni_getfeeshare [ address ] [ ecosystem ]\n"
+            "\nReturns the percentage share of fees distribution applied to the wallet (default) or address (if supplied).\n"
+            "\nArguments:\n"
+            "1. address              (string, optional) retrieve the fee share for the supplied address\n"
+            "2. ecosystem            (number, optional) the ecosystem to check the fee share (1 for main ecosystem, 2 for test ecosystem)\n"
+            "\nResult:\n"
+            "[                       (array of JSON objects)\n"
+            "  {\n"
+            "    \"address\" : nnnnnnn,          (number) the property id\n"
+            "    \"feeshare\" : \"n.nnnnnnnn\",   (string) the percentage of fees this address will receive based on current state\n"
+            "  },\n"
+            "  ...\n"
+            "]\n"
+            "\nExamples:\n"
+            + HelpExampleCli("omni_getfeeshare", "\"1EXoDusjGwvnjZUyKkxZ4UHEf77z6A5S4P\" 1")
+            + HelpExampleRpc("omni_getfeeshare", "\"1EXoDusjGwvnjZUyKkxZ4UHEf77z6A5S4P\", 1")
+        );
+
+    std::string address;
+    uint8_t ecosystem = 1;
+    if (0 < params.size()) {
+        address = ParseAddress(params[0]);
```

Comment:
I found wildcards weren't handled in ```ParseAddressOrEmpty``` so worked around it in https://github.com/OmniLayer/omnicore/pull/359/commits/c40b72baccf93781abe6af718dc62e6e37532f90